File: python/enolib/elements/missing/missing_list.py
from .missing_element_base import MissingElementBase

class MissingList(MissingElementBase):
    def __repr__(self):
        if self._key:
            return f"<class MissingList key={self._key}>"

        return '<class MissingList>'

    def items(self):
        return []

    def optional_string_values(self):
        return []

    def optional_values(self, _loader):
        return []

    def required_string_values(self):
        return []

    def required_values(self, _loader):
        return []

File: ruby/lib/enolib/elements/missing/missing_list.rb
# frozen_string_literal: true

module Enolib
  class MissingList < MissingElementBase
    def items
      []
    end

    def optional_string_values
      []
    end

    def optional_values(_loader)
      []
    end

    def required_string_values
      []
    end

    def required_values(_loader)
      []
    end

    def to_s
      if @key
        "#<Enolib::MissingList key=#{@key}>"
      else
        '#<Enolib::MissingList>'
      end
    end
  end
end
